lib/encoder_unittest.py

```diff
@@ -582,6 +598,37 @@ def testReadResultEncodedInAst(self):
     result = cache.ReadEncodingResult(my_encoding)
     self.assertEquals(result, testresult)
 
+  def testEncodersInMultipleRepos(self):
+    test_tools.EmptyWorkDirectory()
+    context = StorageOnlyContext()
+    cache = encoder.EncodingDiskCache(context)
+    context.cache = cache
+    my_encoder = encoder.Encoder(
+        context,
+        encoder.OptionValueSet(encoder.OptionSet(), '--parameters'))
+    other_dir = encoder_configuration.conf.sysdir() + '/multirepo_test'
+    encoder_configuration.conf.override_scorepath_for_test([other_dir])
+    os.mkdir(other_dir)
```

Comment:
The tempdir is created (and deleted!) by the test superclass - this is a test_tools.FileUsingCodecTest (and that's the main point of the superclass).